### PR TITLE
Fix Darwin build for cmigemo and skk-dict

### PR DESCRIPTION
**Pull request by midchildan**
###### Motivation for this change
Fixes incorrect path to the `iconv` command.

ZHF: #144627

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by jonringer — APPROVED** [review]
diff LGTM

cc @NixOS/darwin-maintainers

File: pkgs/tools/inputmethods/skk/skk-dicts/default.nix [hunk 1]
```diff
@@ -25,6 +25,8 @@ let
     url = "https://raw.githubusercontent.com/skk-dev/dict/8b35d07a7d2044d48b063d2774d9f9d00bb7cb48/SKK-JISYO.assoc";
     sha256 = "1smcbyv6srrhnpl7ic9nqds9nz3g2dgqngmhzkrdlwmvcpvakp1v";
   };
+
+  iconvBin = if stdenv.isDarwin then libiconv else  buildPackages.stdenv.cc.libc;
```
Comment (r-burns) [inline pkgs/tools/inputmethods/skk/skk-dicts/default.nix:29]:
```suggestion
  iconvBin = if stdenv.buildPlatform.isDarwin then buildPackages.libiconv else buildPackages.stdenv.cc.libc;
```

same goes for the other package

**Comment by risicle** [comment]
`nixpkgs-review` happy, macos 10.15

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 146521` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python38Packages.cmigemo</li>
    <li>python38Packages.percol</li>
    <li>python39Packages.cmigemo</li>
    <li>python39Packages.percol</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>cmigemo</li>
    <li>skk-dicts</li>
  </ul>
</details>

**Comment by siraben** [comment]
LGTM on macOS 12.0.1

**Pull request merged**